### ISSUE TITLE
how to run spm_train in multilingual_preprocess.sh

### ISSUE THREAD

**Issue by altctrl00** [cordercorder/nmt-multi#3]
in the _scripts/ted/data_process/multilingual_preprocess.sh_ ,you use spm_train to train spm,is that from fairseq? how can i run it?

**Comment by cordercorder** [comment]
`spm_train` is a command of [sentencepiece](https://github.com/google/sentencepiece#installation) toolkit. Please install [sentencepiece](https://github.com/google/sentencepiece#installation) first and the shell script you mentioned can run well.

**Comment by altctrl00** [comment]
Thanks for your response so quickly, i found i can't install sentencepiece as command line tools because i am not root,i am sorry i am a rookie,is that possible i can install it as a non-root user?

**Comment by cordercorder** [comment]
Yes. You can run `pip install sentencepiece` or `conda install sentencepiece` to install sentencepiece. **After that, the command line tools provided by sentencepiece can be direly used.** There is no need to build and install sentencepiece from the source, which may require root privileges to install the build tools.

**Comment by altctrl00** [comment]
Yes,I have tried `pip install sentencepiece` in my virtual conda env. I activate it and run the command line  `bash multilingual_preprocess.sh`,but it still can not recoginize the lib

**Comment by altctrl00** [comment]
Thanks a lot,  `conda install sentencepiece`  will be a solution, `pip install` may not be  compatible with conda

**Comment by cordercorder** [comment]
There may be some discrepancy between `sentencepiece` from `pip` and `conda`. As `sentencepiece` in my Python environment is installed through `conda install sentencepiece` and the command line tools work well, I thought `pip install sentencepiece` will also work :disappointed_relieved:.

**Comment by altctrl00** [comment]
when I in `nmt-multi` directory to run `bash scripts/ted/data_process/multilingual_preprocess.sh`. It couldn't find `nmt` module in `python -u ${project_dir}/nmt/data_handling/corpus_manager.py` . My project diretory is `/home/.../nmt-multi`.
I was curious that if i add `__init__.py` would work , but it turns to be not.

**Comment by altctrl00** [comment]
In the `data_handling/data_utils`,there is one import `from nmt.tools import Converter`,i couldn't find `nmt.tools`.

**Comment by cordercorder** [comment]
Thanks for reporting these issues.

You can insert the path of `nmt-multi` directory to the environment variable `PYTHONPATH` to make Python interpreter aware of the `nmt` package.  `python -u ${project_dir}/nmt/data_handling/corpus_manager.py` can work well afterward. Below is an example:

``` bash
export PYTHONPATH=/path/to/nmt-multi:${PYTHONPATH}
```

> when I in `nmt-multi` directory to run `bash scripts/ted/data_process/multilingual_preprocess.sh`. It couldn't find `nmt` module in `python -u ${project_dir}/nmt/data_handling/corpus_manager.py` . My project diretory is `/home/.../nmt-multi`. I was curious that if adding `__init__.py` would work , but it turns to be not. I edit the `corpus_manager.py`, changing `nmt.data_handling` to `data_utils` and it can work.


Sorry, this is a mistake during cleaning up the source codes. Please delete this line.
> In the `data_handling/data_utils`,there is one import `from nmt.tools import Converter`,i couldn't find `nmt.tools`.

**Comment by cordercorder** [comment]
Hi, I pushed a new commit to this repository and the changed files can be found at [here](https://github.com/cordercorder/nmt-multi/commit/5eb9f91031c2e4a6075e1573235a4753292247f5). Does this script run well now?
